test/unit/extension/lib/bigcommerce/CartRepositoryTest.js

```diff
@@ -106,4 +106,28 @@ describe('BigCommerceCartRepository - unit', function () {
 
     return subjectUnderTest.getCheckoutUrl().should.eventually.be.rejectedWith(Error)
   })
+
+  it('should properly call bigCommerce client delete method when doing a deleteProductsFromCart', function () {
+    const cartItemIds = ['abc-def-ghi-jk']
+    storageMock.expects('get').once().returns('0000-0000-0000-0000')
+    bigCommerceMock.expects('delete').withArgs('/carts/0000-0000-0000-0000/items/' + cartItemIds[0])
+
+    return subjectUnderTest.deleteProductFromCart(cartItemIds).should.eventually.be.fulfilled
+  })
+
+  it('should see error when bigCommerce delete method throws and error', function () {
+    const cartItemIds = ['abc-def-ghi-jk']
+    storageMock.expects('get').once().returns('0000-0000-0000-0000')
+    bigCommerceMock.expects('delete').withArgs('/carts/0000-0000-0000-0000/items/' + cartItemIds[0]).throws(Error)
+
+    return subjectUnderTest.deleteProductFromCart(cartItemIds).should.eventually.be.rejectedWith(Error)
+  })
+
+  it('should see error when cartId is not returned during delete', function () {
```

Comment:
See previous comment for test naming .

Reply:
Changed name to hopefully better conform the test naming conventions